internal/runtime/converter.go

```diff
@@ -194,8 +194,11 @@ func (c *converter) adjustRuntimeState(dto *pkg.RuntimeDTO) {
 	switch lastOp.State {
 	case string(domain.Succeeded):
 		dto.Status.State = pkg.StateSucceeded
-		if lastOp.Type == pkg.Suspension {
+		switch lastOp.Type {
+		case pkg.Suspension:
 			dto.Status.State = pkg.StateSuspended
+		case pkg.Deprovision:
+			dto.Status.State = pkg.StateDeprovisioning
```

Comment:
In this case the runtime is not in deprovisioning lifecycle state.

IMO the logic should be here:
- `if lastOp.type == Suspension -> StateSuspended`
- if lastOp.type == Deprovision && !dto.Status.DeletedAt.IsZero() -> [`StateDeprovisionIncomplete`](https://github.com/kyma-project/kyma-environment-broker/blob/main/common/runtime/model.go#L26)
- if lastOp.type == Deprovision && dto.Status.DeletedAt.IsZero() -> StateDeprovisioned

Reply:
@vvxxvvxx this also means the mop-agent code must be updated to be able to cope with the [StateDeprovisionIncomplete](https://github.com/kyma-project/kyma-environment-broker/blob/main/common/runtime/model.go#L26) state. In this lifecycle state, the runtime is of course excluded from AvS monitoring.